tests/Unit/Core/Search/Builder/TypeFiltersBuilderTest.php

```diff
@@ -0,0 +1,285 @@
+        $builder->setConfig(['filters_class' => SampleWithConstraintFilters::class]);
+
+        $builder->addTypedBuilder(new SampleFiltersBuilder());
+
+        $builtFilters = $builder->buildFilters();
+        $this->assertEquals(SampleFiltersBuilder::FILTER_ID, $builtFilters->getFilterId());
+        $this->assertEquals(SampleFiltersBuilder::ORDER_BY, $builtFilters->getOrderBy());
+    }
+
+    /**
+     * @param Filters|null $filters
+     *
+     * @return MockObject|FiltersBuilderInterface
+     */
+    private function createBuilderMock(?Filters $filters)
+    {
+        $builderMock = $this
+            ->getMockBuilder(FiltersBuilderInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+
+        $builtFilters = null;
+        if (null !== $filters) {
+            $builtFilters = new SampleWithoutConstraintFilters(
+                array_replace(SampleWithoutConstraintFilters::getDefaults(), $filters->all())
+            );
+        }
+
+        $builderMock
+            ->expects(null !== $filters ? $this->once() : $this->never())
+            ->method('buildFilters')
+            ->willReturn($builtFilters)
+        ;
+
+        return $builderMock;
+    }
+
+    /**
+     * @param array|null $expectedConfig
+     *
+     * @return MockObject|TypedFiltersBuilderInterface
+     */
+    private function createTypeBuilderMock(?array $expectedConfig)
+    {
+        $builderMock = $this
+            ->getMockBuilder(TypedFiltersBuilderInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+
+        $builderMock
+            ->expects(null === $expectedConfig ? $this->never() : $this->once())
+            ->method('supports')
+            ->with($this->equalTo(SampleWithoutConstraintFilters::class))
+            ->willReturn(false)
+        ;
+
+        $builderMock
+            ->expects(null === $expectedConfig ? $this->never() : $this->once())
+            ->method('setConfig')
+            ->withConsecutive(
+                [$this->equalTo($expectedConfig)]
+            )
+        ;
+
+        return $builderMock;
+    }
+}
+
+class SampleWithoutConstraintFilters extends Filters
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getDefaults()
+    {
+        return [
+            'limit' => 42,
+            'offset' => 0,
+            'orderBy' => 'id_sample',
+            'sortOrder' => 'desc',
+            'filters' => [],
+        ];
+    }
+}
+
+class SampleWithConstraintFilters extends Filters
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(array $filters = [], $filterId = '')
+    {
+        if (empty($filterId)) {
+            throw new RuntimeException('Cannot be constructed without filterId');
+        }
+        parent::__construct($filters, $filterId);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getDefaults()
```

Comment:
```suggestion
    public static function getDefaults(): array
```

Reply:
it inherits method from parent and parent doesn't have a type, so we should respect that. Inheritdoc is enough in this case, no?